scalac-scoverage-plugin/src/main/scala/scoverage/plugin.scala

```diff
@@ -153,23 +200,48 @@ class ScoverageInstrumentationComponent(val global: Global, extraAfterPhase: Opt
     def safeSource(tree: Tree): Option[SourceFile] = if (tree.pos.isDefined) Some(tree.pos.source) else None
 
     def invokeCall(id: Int): Tree = {
-      Apply(
-        Select(
+      /**
+       * We still pass in [options.dataDir] as one of the instruments with
+       * [invokedUseEnvironment] as it helps differentiating the source
+       * files at runtime, i.e helps in creating a unique subdir for each source file.
+       */
+      if (options.writeToClasspath){
```

Comment:
The only difference between these two blocks is the `newTermName` value. You should probably extract that into a variable to avoid duplicating the blocks.